src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,137 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.models.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String URL = "jdbc:mysql://localhost:3306/test";
+    private static final String USER = "root";
+    private static final String PASSWORD = "henghfdf";
+
+    @Override
+    public Book create(Book book) {
+        String sqlCommand = "INSERT INTO books (title,price) VALUE (?,?)";
+
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(sqlCommand, Statement.RETURN_GENERATED_KEYS)) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("None of the rows have been added");
+            }
+
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    long id = generatedKeys.getLong(1);
+                    return new Book(id, book.getTitle(), book.getPrice());
+                } else {
+                    throw new RuntimeException("Failed to retrieve the generated key");
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cannot create a connection to the DB", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't add a new book, " + book, e);
```
Check all places where you are using RuntimeEx